### ISSUE TITLE
Regression. Now if an error is thrown in the app.notFound() handler app.onError() doesn't pick it up

### ISSUE THREAD

**Issue by tokenhousetopdev** [honojs/hono#647]
In my app in my not found handler I do this:

```
app.notFound(() => {
  throw new ApiError(httpStatus.NOT_FOUND, 'Not found')
})
```
We can see its not picked up by doing:

```
app.onError(() => {
  console.log('here')
})
```

Previously, this would then forward the error to the error handler but now that isn't the case, can this be fixed please?

**Comment by yusukebe** [comment]
Hi @tokenhousetopdev !

I can't reproduce your problem with the following code:

```ts
const app = new Hono()

app.get('/', (c) => c.text('Hello'))

app.notFound(() => {
  throw new Error('foo!')
})

app.onError((_, c) => {
  console.log('here')
  return c.text('Error', 500)
})
```

If you can, please give me more code or snippets.

**Comment by tokenhousetopdev** [comment]
app.ts:
```
const app = new Hono<{ Bindings: Bindings }>()

app.use('*', cors())

app.notFound(() => {
  throw new ApiError(httpStatus.NOT_FOUND, 'Not found')
})

app.onError((_, c) => {
  console.log('I AM HERE')
  c.status(400)
  return c.text('hi')
})

export default app
```

index.ts:

```
import app from './app'
import './routes/index'

app.fire()
```

I send a post request to a non existant endpoint and I get a 500 error:

```
[mf:err] POST /api/v1/auth/login: Error: Not found
    at Ct.notFoundHandler (/home/repos/personal/fauna-hono-workers-boilerplate/src/app.ts:13:9)
    at /home/repos/personal/fauna-hono-workers-boilerplate/node_modules/hono/dist/hono.js:136:23
    at ServiceWorkerGlobalScope.[kDispatchFetch] (/home/repos/personal/fauna-hono-workers-boilerplate/node_modules/@miniflare/core/src/standards/event.ts:385:13)
    at Server.<anonymous> (/home/repos/personal/fauna-hono-workers-boilerplate/node_modules/@miniflare/http-server/src/index.ts:298:20)
POST /api/v1/auth/login 500 Internal Server Error (152.10ms)
```

**Comment by tokenhousetopdev** [comment]
I am using Hono 2.4.1

**Comment by tokenhousetopdev** [comment]
Other errors are caught and handled as expected just not if I throw an error from the 404 handler

**Comment by yusukebe** [comment]
Hi @tokenhousetopdev 

Sorry, this is just a bug! Fixed in #652 , and it will be released soon. Thank you!
